### PR TITLE
[HWKMETRICS-477] reschedule job for immediate execution if it falls behind

### PR DESCRIPTION
**Pull request by jsanda**
If a job runs long and misses its next execution time, the scheduler will now
execute the job again immediately for the missed execution. It will do so until
the job is caught up.

### CODE REVIEW

File: job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java [hunk 1]
```diff
@@ -165,6 +181,11 @@ public SchedulerImpl(RxSession session) {
         jobFinished = Optional.empty();
     }
 
+    // TODO make configurable
+    private int getQueryThreadPoolSize() {
+        return Runtime.getRuntime().availableProcessors() / 2;
```
Comment (jotak) [inline job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java]: It could return 0, but should be 1 at least... no ?

**Comment by jsanda** [comment]
retest this please

**Comment by jsanda** [comment]
retest this please

**Comment by stefannegrea** [comment]
Thank you @jsanda!

**Pull request merged**